solr/core/src/java/org/apache/solr/servlet/CoordinatorHttpSolrCall.java

```diff
@@ -112,7 +112,22 @@ public static SolrCore getCore(
         core = solrCall.getCoreByCollection(syntheticCollectionName, isPreferLeader);
         if (core != null) {
           factory.collectionVsCoreNameMapping.put(collectionName, core.getName());
-          solrCall.cores.getZkController().getZkStateReader().registerCore(collectionName);
+
+          // only remove on collection deletion, since watch from coordinator is collection specific
+          solrCall
```

Comment:
QQ: this registers watch on state.json file? if so then this must be happening already some other place?

Reply:
Ah ok.got it